### ISSUE TITLE
API to test user access

### ISSUE THREAD

**Issue by codyebberson** [medplum/medplum#2126]
Checking access policies on resources.  For example, "Can this user edit this patient?"

**Comment by codyebberson** [comment]
P1: Client side test
P2: Server side endpoint

**Comment by taylorqj** [comment]
Posted a question relating to this issue in Discord: https://discord.com/channels/905144809105260605/1116045790088003644

This feature will be very useful to us as we're hiding/showing functionality based on the current logged in user's access level. For now, we're fetching `ProjectMembership` and determining if the user's `AccessPolicy` will allow reading/writing on a specific resource. Interested to see where this goes.

In order to achieve the work-around I noticed that the SDK doesn't offer access to the logged in user's id, which is a requirement to query for `ProjectMembership`. I ended up having to decode the access token and using the subject as that's the only place I could find it. I _think_ it might be possible to grab it from initial login, but we're using Google and end up calling `processCode`, which doesn't return any context of the user object.

 I thought that might be useful information for anyone curious or for a future addition!
